src/expire.c

```diff
@@ -533,6 +538,15 @@ void expireGenericCommand(client *c, long long basetime, int unit) {
         addReply(c, shared.cone);
         return;
     } else {
+        /*
+         * If the provided expiry from input was negative, and after
+         * unit conversion and adding `basetime`, if the key was still
+         * not expired, there is an integer overflow.
+         */
+        if (is_expire_input_negative && when > 0) {
```

Comment:
let's convert that one too..
i.e. move the input check to share the same early exit with the other one at the top.